src/ResourceManager/DataMigration/Commands.DataMigration/Cmdlets/NewProjectCmdlet.cs

```diff
@@ -0,0 +1,150 @@
+    [Alias("New-AzureRmDmsProject")]
+    public class NewProjectCmdlet : DataMigrationCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "PSDataMigrationService Object.")]
+        [ValidateNotNull]
+        [Alias("DataMigrationService")]
+        public PSDataMigrationService InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "DataMigrationService Resource Id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "The name of the resource group."
+                )]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "Data Migration Service Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The location of the instance of the Data Migration Service")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the project.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ProjectName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Source platform type for project.")]
+        [ValidateNotNullOrEmpty]
+        public ProjectSourcePlatform SourceType { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Target platform type for project.")]
+        [ValidateNotNullOrEmpty]
+        public ProjectTargetPlatform TargetType { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source Connection Info.")]
+        [ValidateNotNullOrEmpty]
+        public ConnectionInfo SourceConnection { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Target Connection Info.")]
+        [ValidateNotNullOrEmpty]
+        public ConnectionInfo TargetConnection { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Database Infos.")]
+        [ValidateNotNullOrEmpty]
+        public DatabaseInfo[] DatabaseInfo { get; set; }
```

Comment:
[Advisory]: Right now I have to run separate cmd;lets to get database info, target and source connection info in order to run this.  This needs to be simplified.

Reply:
It looks simple object now, but more details will be added in coming releases

Reply:
Yes, the issue is that I should not have to run multiple cmdlets to get the input for this one.  If this is the case, then more parameters or more specialized cmdlets may be necessary.